lib/set-script.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const log = require('npmlog')
+const usageUtil = require('./utils/usage.js')
+const { localPrefix } = require('./npm.js')
+const fs = require('fs')
+const usage = usageUtil('set-script', 'npm set-script [<script>] [<command>]')
+const completion = require('./utils/completion/none.js')
+const parseJSON = require('json-parse-even-better-errors')
+const rpj = require('read-package-json-fast')
+
+const cmd = (args, cb) => set(args).then(() => cb()).catch(cb)
+
+const set = async function (args) {
+  if (process.env.npm_lifecycle_event === 'postinstall') {
+    log.error('Scripts can\'t set from the postinstall script')
```

Comment:
There's no need for both `log.error` and `throw new Error` with the same message, you can remove all the `log.error` calls 😊 

```suggestion
```